.github/workflows/synth.yml

```diff
@@ -0,0 +1,33 @@
+name: Synth
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    types:
+      - opened
+      - edited
+      - synchronize
+      - labeled
+      - unlabeled
+      - reopened
+  workflow_dispatch: {}
+
+jobs:
+  synth:
+    runs-on: ubuntu-latest
+    container:
+      image: docker.mirror.hashicorp.services/hashicorp/jsii-terraform
+    steps:
+      - name: Checkout
+        uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab
+      - name: ensure correct user
+        run: chown -R root /__w/cdktf-integration-serverless-python-gcp-example/
+      - name: Create Yarn Project
+        run: yarn init --yes
+      - name: Add CDKTF CLI
+        run: yarn global add cdktf-cli
```

Comment:
The one thing remaining is that you want to use the same version of `cdktf-cli` that's in your Pipfile so that this "test" stays in sync with the code. As a starting point, check out the regular expression that Daniel figured out here: https://github.com/cdktf/cdktf-integration-serverless-python-gcp-example/blob/36264b461062dcff392ad37cd488cd87ed65e094/scripts/upgrade.sh#L21

(Just the first part, you don't need to do the actual replace part.)

Reply:
[Updated](https://github.com/cdktf/cdktf-integration-serverless-python-gcp-example/pull/9/commits/6b7d5aae4ecfdc0b58925c4f1a4b3da7e5a44ce2#:~:text=%22cdktf_version%3D%24(cat%20Pipfile%20%7C%20grep%20%27cdktf%20%3D%27%20%7C%20awk%20%2DF%27%22%27%20%27%7Bgsub(%22~%3D%22%2C%22%22%2C%242)%3B%20print%20%242%7D%27)%22%20%3E%3E%20%22%24GITHUB_OUTPUT%22) 👍